pychunkedgraph/app/cg_app_blueprint.py

```diff
@@ -242,6 +250,64 @@ def handle_split():
     return app_utils.tobinary(new_roots)
 
 
+@bp.route('/1.0/graph/shatter', methods=['POST', 'GET'])
+def handle_shatter():
+    data = json.loads(request.data)
+
+    user_id = str(request.remote_addr)
+
+    print(data)
+
+    # Call ChunkedGraph
+    cg = app_utils.get_cg()
+
+    data_dict = collections.defaultdict(list)
+
+    k = "sources"
+    node = data[k][0]
+
+    node_id = node[0]
+    radius = node[1]
+    x, y, z = node[2:]
+
+    x /= 2
```

Comment:
You need to document those hacks somewhere, and mark them properly in the code. It's impossible for others to understand this magic.
Applies to all hacks - this here is just the one I am aware of.